bootstrapvz/plugins/puppet/tasks.py

```diff
@@ -36,28 +52,75 @@ def run(cls, info):
             raise TaskError(msg)
 
 
-class AddPackages(Task):
-    description = 'Add puppet package'
+class AddPuppetlabsPC1SourcesList(Task):
+    description = 'Adding Puppetlabs APT repo to the list of sources.'
     phase = phases.preparation
 
     @classmethod
     def run(cls, info):
-        info.packages.add('puppet')
+        if (info.manifest.release == jessie):
+            info.source_lists.add('puppetlabs', 'deb http://apt.puppetlabs.com jessie PC1')
+        if (info.manifest.release == wheezy):
+            info.source_lists.add('puppetlabs', 'deb http://apt.puppetlabs.com wheezy PC1')
+
+
+class InstallPuppetlabsPC1ReleaseKey(Task):
+    description = 'Install puppetlabs PC1 Release key into the keyring'
+    phase = phases.package_installation
+
+    @classmethod
+    def run(cls, info):
+        key_file_name = 'DEB-GPG-KEY-puppet'
```

Comment:
I was just wondering, why not just do like [`InstallTrustedKeys`](https://github.com/andsens/bootstrap-vz/blob/7af9cd9fb85d845884b2bd031fa694cc3c101d81/bootstrapvz/common/tasks/apt.py#L111) does?
i.e. `git add` the key to `plugins/puppet/assets/gpg-key` and then install it by copying:
```
from shutil import copy
key_path = os.path.join(ASSETS_DIR, 'DEB-GPG-KEY-puppet')
destination = os.path.join(info.root, 'etc/apt/trusted.gpg.d/DEB-GPG-KEY-puppet')
copy(key_path, destination)
```

where `ASSETS_DIR` is define in the top like in the [expand_root plugin](https://github.com/andsens/bootstrap-vz/blob/7af9cd9fb85d845884b2bd031fa694cc3c101d81/bootstrapvz/plugins/expand_root/tasks.py#L11). Like so: `ASSETS_DIR = rel_path(__file__, 'assets')`

This way you avoid running a load of commands and you don't install `ca-certificates`, which the user didn't specify as a package really.

Sorry for only seeing this now :-/